_posts/2019-11-08-quarkus-for-spring-developers.adoc

```diff
@@ -0,0 +1,58 @@
+
+image::TraditionalStack.png[Traditional Java Stack, 450]
+
+This is the very problem that Quarkus addresses.  Quarkus is optimized for memory density and rapid startup time. Quarkus applications running on the JVM can deliver nearly twice as many application instances in the same amount of RAM when compared to other cloud native Java stacks, and up to 7x more instances when packaged as a native binary. This is more than simply compiling to a native binary using https://github.com/oracle/graal/tree/master/substratevm[SubstrateVM] (a feature of https://www.graalvm.org/[GraalVM]). Quarkus has optimized traditionally “highly dynamic” frameworks for Kubernetes immutable infrastructure, resulting in reduced memory utilization and faster initialization. These optimized and https://quarkus.io/guides/[well-documented] frameworks, called “extensions”, consist of best of breed and standard APIs. The result is https://quarkus.io/assets/images/quarkus_metrics_graphic_bootmem_wide.png[significant improvements in runtime efficiency].
+
+image::QuarkusStack.png[Quarkus Stack, 450]
+
+== What existing knowledge can Spring developers bring to Quarkus?
+Quarkus’ Spring API compatibility includes https://quarkus.io/guides/spring-di-guide[Spring DI], https://quarkus.io/guides/spring-web-guide[Spring Web], and https://quarkus.io/guides/spring-data-jpa-guide[Spring Data JPA]. Additional Spring APIs are being planned like Spring Security and Spring Config. When running on the JVM, Quarkus applications can utilize virtually any Java library. Many libraries that are not dynamic in nature (do not use Java refection) will also compile to native. Lombok, a popular library among Spring developers for example, works with native compilation. To be clear, the Spring API support in Quarkus is not intended to be a complete Spring platform to re-host existing Spring applications. The intent is to offer enough Spring API support to make developing new applications with Quarkus a natural getting started experience. When combined with pre-optimized extensions, Quarkus delivers an https://quarkus.io/guides/[amazing amount of functionality] for microservices development. With all this being said, developers have successfully migrated Spring applications to Quarkus :-)
+
+[IMPORTANT]
+====
+The Spring Framework is highly dynamic in nature. To address this, the Quarkus Spring compatibility extensions map Spring APIs to APIs in existing extensions that have already been optimized for fast startup, reduced memory utilization and native compilation, like RestEasy and CDI. Also, the Quarkus Spring compatibility extensions do not utilize the Spring application context. For these reasons, attempting to utilize additional Spring libraries will likely not work.
+====
+
+*Quarkus Spring Web Example*
+
+image::QuarkusSpringWeb.png[Quarkus Spring Web, 600]
```

Comment:
instead of images you could just include the code like,

```
[source,java]
----
package ...

----
```

That would be nicer and copy/pasteable :)

Reply:
Fixed. I was wondering how to do that :-)